oe_webtools_analytics/src/EventSubscriber/AnalyticsEventSubscriber.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * @file
+ * Listening to the AnalyticsEvent.
+ */
+
+namespace Drupal\oe_webtools_analytics\EventSubscriber;
+
+use Drupal\oe_webtools_analytics\AnalyticsEventInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Drupal\oe_webtools_analytics\Event\AnalyticsEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Event Subscriber AnalyticsEventSubscriber.
+ */
+class AnalyticsEventSubscriber implements EventSubscriberInterface {
+  /**
+   * The Configuration overrides instance.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * AnalyticsEventSubscriber constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The configuration object.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+   *   The request on the stack.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, RequestStack $requestStack) {
+    // Get id from settings.php!
+    if (!$configFactory->get('oe_webtools.analytics')) {
```

Comment:
This statement is always true, the get function always returns a Config object so we will never trigger the exception. You should instead try check against the raw values or against an specific value in the config. On the same note, its strange that here we trigger a full blown exception when the settings are not there but on the requirements hook we just give a warning. Shouldn't we prevent the usage of the module alltogether if the configuration is not in place?